metadata/metadata.go

```diff
@@ -82,6 +83,42 @@ func Pairs(kv ...string) MD {
 	return md
 }
 
+// Validate returns error if md is not valid
+// There are check items:
+// - header names contain one or more characters from this set [0-9 a-z _ - .]
+// - if the header-name ends with a "-bin" suffix, the header-value could contain an arbitrary octet sequence. So no real validation required here.
+// - if header-name does not end with a "-bin" suffix, header-value should only contain one or more characters from the set ( %x20-%x7E ) which includes space and printable ASCII.
+func (md MD) Validate() error {
+	for k, vals := range md {
+		// check key
+		for _, rc := range []rune(k) {
```

Comment:
`for i := 0; i < len(k); i++ {}` and use `k[i]` _might_ be slightly faster (it saves a conversion, maybe?)

Reply:
Maybe `for _, rc := range k` is DRYer and saves a conversion?

Reply:
Range behaves differently for strings: https://go.dev/play/p/2B-0HiWektw

If I understand this correctly, `for range` converts (and merges) bytes to runes, but `for i++` doesn't do that.